backend/src/main.rs

```diff
@@ -23,19 +19,29 @@ async fn board_record_move(path: web::Path<(u32, String)>) -> impl Responder {
     let (board_id, board_move) = path.into_inner();
     println!("board_id: {}, move: {}", board_id, board_move);
     let game = "game.txt";
-    let history = History::from_filepath(game);
+    let mut history;
+    match History::from_filepath(game) {
+        Ok(h) => history = h,
+        Err(e) => return HttpResponse::BadRequest().body(format!("{}", e)),
+    };
     println!("{:?}", history);
-    let mut state = State::new_from_history(&history);
+    let mut state;
+    match State::new_from_history(&history) {
+        Ok(s) => state = s,
+        Err(e) => return HttpResponse::BadRequest().body(format!("{}", e)),
+    };
     // TODO this is hacky af
     let tokens = board_move.split_whitespace().collect::<Vec<&str>>();
     let piece = *tokens.get(0).unwrap();
     let position = *tokens.get(1).unwrap();
-    state.play_turn_from_notation(piece, position);
+    match state.play_turn_from_notation(piece, position) {
+        Ok(_) => {},
+        Err(e) => return HttpResponse::BadRequest().body(format!("{}", e)),
+    };
     state.history.write_move(game, state.turn, board_move);
-    println!("{}", state.board);
-    HttpResponse::Ok()
+    HttpResponse::Ok().into()
```

Comment:
is the `into()` needed here?

Reply:
I'm joining the convo to hopefully learn something as I'm kinda clueless. 

Without the into we get mismatched types, expected actix_web::HttpResponse, found 'HttpResoonseBuilder' is there a better way to resolve this than the into?

Reply:
ah you're right! i think `into()` is fine in this case, although another option could be to use [`finish()`](https://docs.rs/actix-web/4.3.1/actix_web/struct.HttpResponseBuilder.html#method.finish), which is a little more explicit about what type conversion is happening here. i could go either way.

also if you'd like to learn more about HttpResponse vs HttpResponseBuilder, the docs have a bit on it: https://actix.rs/docs/response/

Reply:
that works :slightly_smiling_face: